### PR TITLE
Fix native go build

### LINKED ISSUE #5602: Inconsistent path for node-labeller.sh in virt-handler image
/kind bug

**What happened**:

node-labeller.sh is packaged into the /usr/bin directory as shown here https://github.com/kubevirt/kubevirt/blob/e286b55c2c51d7dd7afda162722b142103df3968/cmd/virt-launcher/BUILD.bazel#L145

but is referred by /bin/node-labeller.sh when it is being used here https://github.com/kubevirt/kubevirt/blob/e286b55c2c51d7dd7afda162722b142103df3968/pkg/virt-operator/resource/generate/components/daemonsets.go#L83

**What you expected to happen**:

They should be consistent.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`): 0.40.0
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by vasiliy-ul**
**What this PR does / why we need it**:

The PR includes several fixes for native Go build (without Bazel):
- drop deprecated `-i` go build flag (to suppress warning)
- fix inconsistent path for node-labeler.sh (issue https://github.com/kubevirt/kubevirt/issues/5602)
- allow linux-only builds of virtctl (without windows and darwin)
- allow using go build for the templator when generating the manifests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5602

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/build-go.sh [hunk 1]
```diff
@@ -116,17 +116,17 @@ for arg in $args; do
...
             (cd ${CMD_OUT_DIR}/${BIN_NAME} && ln -sf ${LINUX_NAME} ${BIN_NAME})
 
             kubevirt::version::get_version_vars
             echo "$KUBEVIRT_GIT_VERSION" >${CMD_OUT_DIR}/${BIN_NAME}/.version
 
             # build virtctl also for darwin and windows on amd64
-            if [ "${BIN_NAME}" = "virtctl" -a "${ARCH}" = "amd64" ]; then
-                GOOS=darwin GOARCH=amd64 go_build -i -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCH_BASENAME}-darwin-amd64 -ldflags "$(kubevirt::version::ldflags)" $(pkg_dir darwin amd64)
-                GOOS=windows GOARCH=amd64 go_build -i -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCH_BASENAME}-windows-amd64.exe -ldflags "$(kubevirt::version::ldflags)" $(pkg_dir windows amd64)
+            if [ "${BIN_NAME}" = "virtctl" -a "${ARCH}" = "amd64" -a -z "${LINUX_ONLY}" ]; then
```
Comment (dhiller) [inline hack/build-go.sh:127]: Where does `LINUX_ONLY` come from?

Comment (vasiliy-ul) [reply hack/build-go.sh:127]: In our case we invoke the script directly. So I was thinking about smth like `LINUX_ONLY=1 ./hack/build-go.sh ...`.

**Review by dhiller — APPROVED** [review]
Thanks for the fixes

/lgtm

One question I have still: what is the use case for injecting a custom manifest-templator?

**Comment by vasiliy-ul** [comment]
> One question I have still: what is the use case for injecting a custom manifest-templator?

@dhiller , its not for a 'custom' templator. The issue for us is that `build-manifests.sh` triggers the build of the templator using `bazel`: https://github.com/kubevirt/kubevirt/blob/37ec453d972a7b4957fb6c1160cfdba71d007512/hack/build-manifests.sh#L35-L37

Unfortunately bazel does not work well in our build environemnt as it tends to pull many dependencies from the web (which we do not need in general since we provide our own containers). The idea to work it around is to prebuilt the default templator using native go `./hack/build-go.sh install  tools/manifest-templator` and then call

```
MANIFEST_TEMPLATOR=./_out/cmd/manifest-templator/manifest-templator ./hack/build-manifests.sh
```

**Comment by rmohr** [comment]
> Unfortunately bazel does not work well in our build environemnt as it tends to pull many dependencies from the web (which we do not need in general since we provide our own containers)

It may probably still not be the best option for you to build, but note that offline building would be possible by pre-fetching dependencies and pointing bazel to directory with them via `--distdir`.

**Comment by vasiliy-ul** [comment]
> 
> It may probably still not be the best option for you to build, but note that offline building would be possible by pre-fetching dependencies and pointing bazel to directory with them via `--distdir`.

Thanks for the suggestion. I've already looked a bit into that direction. Indeed it is possible to setup offline builds with bazel. But in fact we do not need most of the dependencies like e.g. RPMs and exteranl modules since at the end we build our own container images. We really need only the binaries and currently the easiest way to get those is by using native go build.

File: hack/build-manifests.sh [hunk 2]
```diff
@@ -28,13 +28,16 @@ kubevirt_logo_path="assets/kubevirt_logo.png"
...
 rm -rf ${MANIFEST_TEMPLATES_OUT_DIR}
 
-rm -rf "${TESTS_OUT_DIR}/tools"
-mkdir -p "${TESTS_OUT_DIR}/tools"
-templator=${TESTS_OUT_DIR}/tools/manifest-templator
-
-bazel run \
-    --config=${HOST_ARCHITECTURE} \
-    //:build-manifest-templator -- ${templator}
+templator=${MANIFEST_TEMPLATOR}
```
Comment (rmohr) [inline hack/build-manifests.sh:31]: A clear indicator if one wants to use `bazel` or `go` would be something I would prefer. It a little bit a mix-and-match with unclear fallback paths for some users. Could you improve that a little?

**Review by rmohr — CHANGES_REQUESTED** [review]

File: hack/build-manifests.sh (hunk 2 above)
Comment (vasiliy-ul) [reply hack/build-manifests.sh:31]: I was thinking about moving the build of `manifest-templator` tool to a separate make target. Or even better: include it in the `bazel-build` / `go-build`. That way `hack/build-manifests.sh` can only do the manifests generation. What do you think? Is there any reason to keep `bazel run ... //:build-manifest-templator ...` here?

**Review by dhiller — APPROVED** [review]

**Pull request merged**